pyfr/integrators/controllers.py

```diff
@@ -49,6 +52,14 @@ def __init__(self, *args, **kwargs):
         # Delete the memory-intensive elements map from the system
         del self.system.ele_map
 
+    def set_dt(self, dt):
+        tlist = [self.tlist, np.arange(self.tcurr, self.tend, dt)]
+
+        # List of sorted unique times
+        digits = int(np.log10(1.0/self.dtmin))
+        self.tlist = np.sort(np.unique(np.around(np.concatenate(tlist),
```

Comment:
If you want to see if two things are within a tolerance of each other rounding is not the correct solution.  A good example of this is in the fuzzy sorting code.  A better solution is along the lines of:

```
for t in np.arange(self.tcurr, self.tend, dt):
    if all(abs(s - t) >= self.dtmin for s in self.tlist):
        self.tlist.append(s)
        
self.tlist.sort()
```
It is also quite easy to extend the method to accept a list of times as opposed to a step.